src/robot/libraries/Collections.py

```diff
@@ -842,16 +900,19 @@ def _keys_should_be_equal(self, dict1, dict2, msg, values):
         _verify_condition(not error, '\n'.join(error), msg, values)
         return keys1
 
-    def _key_values_should_be_equal(self, keys, dict1, dict2, msg, values):
-        diffs = '\n'.join(self._yield_dict_diffs(keys, dict1, dict2))
+    def _key_values_should_be_equal(self, keys, dict1, dict2, msg, values, ignore_case):
+        normalize = Normalizer(ignore_case).normalize
+        diffs = '\n'.join(self._yield_dict_diffs(normalize(dict1),
+                                normalize(dict2), ignore_case))
         _verify_condition(not diffs,
                           f'Following keys have different values:\n{diffs}',
                           msg, values)
 
-    def _yield_dict_diffs(self, keys, dict1, dict2):
-        for key in keys:
+    def _yield_dict_diffs(self, dict1, dict2, ignore_case):
+        normalize = Normalizer(ignore_case).normalize
+        for key1, key2 in zip(normalize(dict1), normalize(dict2)):
             try:
-                assert_equal(dict1[key], dict2[key], msg=f'Key {key}')
+                assert_equal(dict1[key1], dict2[key2], msg=f'Key {key1}')
```

Comment:
This implementation isn't correct and causes a failure in an unrelated suite that uses `Dictionaries Should Be Equal` with dicts having items in different order. The old `for key in keys` approach avoided the issue and should be restored. `normalize` can then be used with each value like `assert_equal(normalize(dict1[key]), ...)`.

Reply:
I fixed this, by resetting it to for key in keys, so this behaviour is fixed. Now differently ordered dictionaries also work. I also added some tests for this. I did this by change it to `for key in keys:`

I do however now encounter a problem at various tests because of key errors, that I haven't yet solved. Those can be solved by changing it to `for key in normalize(keys):`, but this poses a new problem as keys is list-like, meaning it will always normalize, unless `ignore_case=False`. I will work on this some more when I have time tomorrow

Reply:
I have now also fixed the error mentioned above in a very ugly way. I will commit it for now but put in some more time tomorrow to see if a cleaner solution is possible

Reply:
Right, if you have `{'A': 1, 'B': 2}` and `{'a': 1, 'b': 2}`, keys are equal case-insensitively, but comparing values by iterating over keys of either of the dicts and doing `d1[k] == d2[k]` fails. Keys are returned by the method comparing keys, and it could possibly return a list of two-tuples containing matching keys of both of the dicts.

Reply:
Things get even more complicated if you have `{'a': 1, 'A': 2}` and `{'a': 2, 'A': 1}`. Keys are equalnand I guess we can consider values to match case-insensitively as well. Value comparison will fail, however, if comparison is done using key pairs `[('a', 'a'), ('A', 'A')]`. This gets even more "fun" with dicts having keys like `'abc', 'Abc', 'aBc', 'abC', 'ABc', 'AbC', 'aBC', 'ABC'` so that they have same values but with different keys. I have and idea how to handle this in ganeric manner, but I'm currently on mobile and writing code examples is too hard. I probably should test my idea as well.

Reply:
Hmm, that would be a solution, but we don't know the actual pairs. The current `_keys_should_be_equal` just checks if the same keys exist in the dictionaries.

Reply:
I made an example using `keys = [(key1, key2) for key1, key2 in
                zip(dict1.keys(), dict2.keys()) if normalize(key1) == normalize(key2)]`
It works, however, now a new challenge rises when amount of keys is not equal. So, still ongoing

Reply:
Ah, I found it. In `dictionary_should_contain_sub_dictionary` we also create a set of keys. This should return the couples as well

Reply:
This doesn't solve the problem yet for keys with double names. But I think that might be something that can't be solved. Because you don't know the actual pairs. If i create 2 dicts:

```
dict1 = {'abc': 3, 'AbC':4}
dict2 = {'ABC':4, 'Abc': 3}
```
It is now impossible to check if this is correct. I have 2 keys that are 3 and 2 that are 4, but i don't actually know if they are similar. Especially if they are also in a different order.

I think that if you have the identically named keys (except for case), that you shouldn't be able to use ignore_case. So we could add an error message when multiple keys are present.

Reply:
I forgot to test it, but I believe this algorithm would solve the above problem:
1. Collect all values with matching keys to lists. In the above case we'd get lists `[3, 4]` and [4, 3]`.
2. Validate that lists have same items.This wouldn't be that easy because we cannot simply sort lists because they could be unsortable nor could we use sets because values aren't necessarily hashable.

Even though we could make the above work, I'm not sure is it worth the effort or a good idea in general. Making it an error to have multiple keys that normalize to the same value could be better.

Reply:
I have now added an error in `_keys_should_be_equal`. I used:
`len(set(normalize(dict1).keys()))!=len(dict1.keys())`
I think this is valid, since normalize changes dict_like objects to a dictionary. So I can use sets too, after the normalizing